STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/_az_nx_blob_client.c

```diff
@@ -0,0 +1,148 @@
+#include "_az_nx_blob_client.h"
+
+#define USER_AGENT_NAME "User-Agent: "
+#define USER_AGENT_VALUE "Azure RTOS Device (STM32)"
+
+static az_result result_from_nx_status(UINT nx_status)
+{
+  switch (nx_status)
+  {
+    case NX_SUCCESS:
+      return AZ_OK;
+    case NX_WEB_HTTP_POOL_ERROR:
+      return AZ_ERROR_NOT_ENOUGH_SPACE;
+    case NX_NO_PACKET:
+      return AZ_ERROR_NOT_ENOUGH_SPACE;
+    case NX_INVALID_PARAMETERS:
+      return AZ_ERROR_NOT_SUPPORTED;
+    case NX_OPTION_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_PTR_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_CALLER_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_WEB_HTTP_NOT_READY:
+      return AZ_ERROR_ULIB_BUSY;
+    case NX_WAIT_ABORTED:
+      return AZ_ERROR_ULIB_SYSTEM; // should be AZ_ERROR_ULIB_TIME_OUT
+                                   // or AZ_ERROR_ULIB_ABORTED
+    default:
+      return AZ_ERROR_ULIB_SYSTEM;
+  }
+}
+
+az_result az_nx_blob_client_init(
```

Comment:
may want AZ_NODISCARD for each function here. @mamokarz thoughts?

Reply:
Yes, it is a good idea to make all public APIs as AZ_NODISCARD.

Reply:
these will not be public but I will make them AZ_NODISCARD anywas.